Migrate vof example (#350)
lethe-cfd/lethe: doc/source/examples/multiphysics/dam-break-VOF/dam-break-VOF.rst

==========================
Dam break VOF
==========================

> 

Reviewer: Maybe mention the features of the example (similar to 2D Lid-Driven Cavity Flow example). You can mention adaptive time-stepping method as a feature, for instance.

Reviewer 2: +1 Please do so.

This example simulates the dam break experiments of `Martin and Moyce (1952)`_. 

.. _Martin and Moyce (1952): https://royalsocietypublishing.org/doi/abs/10.1098/rsta.1952.0006

A liquid is fixed behind a dam at the left most corner of
a rectangular domain as shown in the figure below.

> At :math:`t = 0` seconds, the dam is removed, and 

Reviewer: ```suggestion
At :math:`t = 0` s, the dam is removed, and 
```

the liquid is released into the total simulation domain. 
The corresponding parameter file is 

> ``gls_VOF_dam-break_Martin_and_Moyce.prm``.

Reviewer: Add the location of the example prm file. Look at the 2D Lid-Driven Cavity Flow example (location of the example).

Author: I don't think that that is necessary. At this point anyone who knows how to move around the documentation tree knows how to get to an example

> The following schematic describes the simulation:

Reviewer: ```suggestion
The following schematic describes the geometry and dimensions of the simulation:
```

.. image:: images/VOF_dam_break_configuration.png
    :alt: Schematic
    :align: center

.. note:: 

>     All the four boundary conditions are slip, and an external 

Reviewer: ```suggestion
    All the four boundary conditions are ``slip``, and an external 
```

Author: I don't understand this one

Reviewer 2: ``slip`` instead of slip will write it in monospace to reference the keyword of the boundary condition.

>     gravity field of :math:`-1`` is applied in the y direction.

Reviewer: ```suggestion
    gravity field of :math:`-1`` is applied in the `y` direction.
```

--------------
Parameter file
--------------

Time integration is handled by a 1st order backward differentiation scheme 
`(bdf1)`, for a :math:`4.1` seconds simulation time with an initial 
time step of :math:`0.01` seconds.

.. note::   
    This example uses an adaptive time-stepping method in which the 

>     time-steps are modified during the simulation to keep the max CFD 

Reviewer 2: ```suggestion
    time-steps are modified during the simulation to keep the maximum value of the CFL condition
```

    below a given threshold.

.. code-block:: text

    # --------------------------------------------------
    # Simulation Control
    #---------------------------------------------------
    subsection simulation control
        set method                         = bdf1
        set time end                       = 4.1
        set time step                      = 0.01
        set adapt                          = true
        set max cfl                        = 0.5
        set stop tolerance                 = 1e-5
        set adaptative time step scaling   = 1.1
        set number mesh adapt              = 0
        set output name                    = dam-break_VOF
        set output frequency               = 20
        set output path                    = ./Output/
        set subdivision                    = 1      
    end

> Make sure to create a directory named `Output` in the same directory 

Reviewer: ```suggestion
Make sure to create a directory named ``Output`` in the same directory 
```

> Make sure to create a directory named `Output` in the same directory 
> you are calling the solver from.   

Reviewer 2: ```suggestion
.. warning::
Make sure to create a directory named `Output` in the same directory 
you are calling the solver from.  Otherwise, the solver will be unable to generate the results files and will hang.
```

The ``multiphysics`` subsection enables to turn on `(true)` 
and off `(false)`` the physics of interest. Here ``VOF`` and 
``interface sharpening`` are chosen.

> .. note:: 

Reviewer: ```suggestion
.. warning:: 
```

     If the interface sharpening is not selected, the interface 
     between phases will become blurry (due to diffusion). 

>      The following section defines the interface sharpening 

Reviewer 3: ```suggestion
     The next section defines the interface sharpening 
```

     parameters.

.. code-block:: text

    #---------------------------------------------------
    # Multiphysics
    #---------------------------------------------------
    subsection multiphysics
        set VOF = true
        set interface sharpening = true
    end 

> In the ``interface sharpening`` subsection, the parameters required for 

Reviewer 3: "interface sharpening" text should be a html link to the 2-Interface sharpening section of the page, not a code block

Reviewer 3: forget it, see next comment

sharpening the interface are defined. The current interface 
sharpening method consists of two steps. These steps are explained as
follows: 

""""""""""""""""""""""""""

> 1- Phase fraction limiter

Reviewer 3: not sure it should be a title here. Maybe a numbered list. I would rearrange this a little :

Remove text "In the interface sharpening subsection, the parameters required for sharpening the interface are defined. The current interface sharpening method consists of two steps. These steps are explained as follows:"

Replace by:
""""""""""""""""""""""""""
Interface Sharpening Parameters
""""""""""""""""""""""""""
The current interface sharpening method consists of two steps: 
1. Phase fraction limiter (not a title, but a numbered list)
(equation)
2. Sharpening
(equation)

""""""""""""""""""""""""""

.. math:: 

>     \phi := min \left( max \left(\phi^{old},0 \right),1 \right)

Reviewer 2: Can you add a bit of a description as to what is achieved here?

>     \phi := min \left( max \left(\phi^{old},0 \right),1 \right)

Reviewer: ```suggestion
    \phi = min \left( max \left(\phi^{old},0 \right),1 \right)
```

""""""""""""""""""""""""
2 -Interface sharpening 
""""""""""""""""""""""""
.. math::
    \phi :=
    \begin{cases}
     c^{1-\alpha} \phi^{\alpha} &  (0 \leq \phi < c  ) \\
     1-(c-1)^{1-\alpha}(1-\phi)^{\alpha} & (c \leq \phi \leq 1  ) 
    \end{cases}

where :math:`\phi`, :math:`c`, and :math:`\alpha` denote phase fraction, 
sharpening threshold, and interface sharpness respectively. 
This interface sharpening method was proposed by `Aliabadi and Tezduyar (2000)`_.  

.. _Aliabadi and Tezduyar (2000):  https://www.sciencedirect.com/science/article/pii/S0045782500002000

> `Sharpening frequency` is an integer parameter that defines the 
> frequency of interface sharpening; sharpening threshold defines 
> a phase fraction threshold for interface sharpening (generally :math:`0.5`);
> and interface sharpness is a model parameter which is generally in
> the range of :math:`(1-2]`.

Reviewer 2: ```suggestion
``Sharpening frequency`` is an integer parameter that defines the 
frequency of the interface sharpening; ``sharpening threshold`` defines 
a phase fraction threshold for interface sharpening (generally :math:`0.5`);
and ``interface sharpness`` is a model parameter which is generally in
the range of :math:`(1-2]`.
```

Can you give guidelines on the interface sharpness parameter? What is a good default value for it.

.. code-block:: text

    #---------------------------------------------------

>     # Interphase sharpening

Reviewer 3: Interface

Reviewer 3: ```suggestion
    # Interface sharpening
```

Reviewer 3: and change that in the .prm please ;)

    #---------------------------------------------------
    subsection interface sharpening
        set sharpening threshold        = 0.5
        set interface sharpness         = 2
        set sharpening frequency      	= 10
    end

> In the ``initial condition``, the initial velocity and initial position 

Reviewer 3: Consequently to the Interface sharpening subsection, add before this line:
""""""""""""""""""""""""""
Fluid phase parameters
""""""""""""""""""""""""""

of the liquid phase are defined. The liquid phase is initially 

> defined as rectangle of length :math:`= 3.5` and height ::math:`= 7`.

Reviewer 2: Can you add an explanation on how the liquid is defined as 1 and the gas is defined as 0 in this case.

Reviewer 3: ::math:`= 7` remove one left ":"

    subsection source term
        set enable = true
        subsection xyz
            set Function expression = 0;-1.0; 0
        end
    end

Two fluids are present in this simulation, hence in the ``physical 
properties`` subsection, their physical properties should be defined:

.. code-block:: text

    #---------------------------------------------------
    # Physical Properties
    #---------------------------------------------------
    subsection physical properties
        set number of fluids     = 2
        subsection fluid 0
            set density              = 0.02
            set kinematic viscosity  = 0.1
        end
        subsection fluid 1
            set density              = 0.9982
            set kinematic viscosity  = 0.01
        end
    end

In the ``mesh adaptation subsection``, adaptive mesh refinement is 
defined for ``velocity``. ``min refinement level`` and ``max refinement 
level`` are 4 and 5, respectively.

.. code-block:: text

    #---------------------------------------------------
    # Mesh Adaptation
    #---------------------------------------------------
    subsection mesh adaptation
        set type                    = kelly

>         set variable                = velocity

Reviewer 3: @shahabgol maybe do it with `variable = phase` instead, if you want to refinement to track the interface. For my cases I have a very good visualization if I set `fraction coarsening = 0.1` and `fraction refinement = 0.8` (we refine 80% of cells with the highest Kelly estimator for the phase)

Reviewer 2: Yeah the best way it definitely to use set variable = phase

        set fraction type           = fraction

>         set max refinement level    = 5

Reviewer 3: we are missing the `mesh` subsection of the .prm to understand that (initial refinement = 4, so you could put max refinement level to 6 and min level to 3 I think)

Reviewer 3: still to be addressed ;)

Author: Attention to detaal

        set min refinement level    = 4
        set frequency               = 5
        set fraction refinement     = 0.2

>         set fraction coarsening     = 0.01

Reviewer 2: That's a very low coarsening fraction, is there a reason for this? You will be keep a lot of elements for no reason

Reviewer: @AbbasBalloutRe Can you try simulating the example with different values of fraction coarsening? I'm not sure but I think when I launched it with larger coarsening values, the refined cells did not cover the whole interface.

    end

*Call the gls_navier_stokes_2d by invoking:*  

> ``mpirun -np 2 ./{path-to-lethe-build-dir}/applications/gls_navier_stokes_2d/gls_navier_stokes_2d gls_VOF_dam-break_Martin_and_Moyce.prm``

Reviewer: ```suggestion
``mpirun -np 2 gls_navier_stokes_2d gls_VOF_dam-break_Martin_and_Moyce.prm``
```

> ``mpirun -np 2 ./{path-to-lethe-build-dir}/applications/gls_navier_stokes_2d/gls_navier_stokes_2d gls_VOF_dam-break_Martin_and_Moyce.prm``

Reviewer 3: make it a block code rather than an inline code

> *to run the simulation using two CPU cores.* (Feel free to use more)

Reviewer 3: remove italic

.. warning:: 
    The code will compute :math:`100,000+` dofs for :math:`620+` time 
    iterations. Make sure to compile lethe in `Release` mode and 
    run in parallel using mpirun 

-------
Results
-------
The following image shows the screenshots 
of the simulation at :math:`0`, :math:`1.1`, :math:`3`, and :math:`4` seconds

.. image:: images/time-shots.png
    :alt: time-shots
    :align: center

A python post-processing code `(gls_VOF_dam-break_Martin_and_Moyce.py)` 

> is added to the example folder for post-processing the results.

Reviewer 2: ```suggestion
is added to the example folder to post-process the results.
```

Run `python3 ./Dambreak_2d_lethe.py ./Output` to execute this 
post-processing code, where `./Output` is the directory that 
contains the simulation results. In post-processing, the maximum 
dimensionless lateral position of the liquid phase is tracked 
through time and compared with the experiments of Martin and Moyce
(1952). The following figure shows the result of
the post-processing:

.. image:: images/xmax_t.png
    :alt: xmax_t
    :align: center

As mentioned previously, this simulation uses adaptive mesh
refinement. The following image shows the mesh and the position of
the interface at :math:`4` seconds. The mesh refinement detects 

> and refines the meshes on the interface.

Reviewer 2: Clearly the coarsening is not agressive enough, you see cells remaining :).

Reviewer 3: and with refinement with the phase parameter it will be even more beautiful :)

Reviewer: Refinement with the phase fraction does not work in this example. It does not refine the moving side of the interface. :)

Reviewer 3: Have you tried with   
`set fraction type		= fraction`
`set fraction refinement = 0.8`

? (see the new doc https://lethe-cfd.github.io/lethe/parameters/cfd/mesh_adaptation_control.html ;) )

Reviewer 3: if you put `set fraction refinement = 0.1` as you would for a velocity mesh adaptation, it is normal that it does not want to refine the most "blurry" parts of the interface, which are the moving parts

Reviewer: yes, it still does not work with fraction refinement = 0.8 :)
![image](https://user-images.githubusercontent.com/29518349/150181216-b4f90552-1026-4a18-baaa-f0100ccd0099.png)

Reviewer 2: In this case you can try 0.9 or 0.95 for example, anyway the rest of the cells will be far below this margin of error.

Reviewer 3: I agree that it needs time to catch up with fast moving interface, but it still is working properly. I think you base mesh is too thin for you to really see it, so here is what I have with these parameters (much coarser mesh at the beginning, so really quicker simulation):

`#---------------------------------------------------
# Mesh
#---------------------------------------------------
subsection mesh
        set type = dealii
        set grid type = subdivided_hyper_rectangle
        set grid arguments = 14, 10 : 0, 0 : 14, 10 : true
        set initial refinement = 3
end

#---------------------------------------------------
# Mesh Adaptation
#---------------------------------------------------
subsection mesh adaptation
  set type                    = kelly
  set variable                = phase
  set fraction type           = fraction
  set max refinement level    = 5
  set min refinement level    = 2
  set frequency               = 1
  set fraction refinement     = 0.9
  set fraction coarsening     = 0.1
end`

At iteration 0:
![image](https://user-images.githubusercontent.com/72091781/150186645-1a60c8b9-d10a-4da6-9a5a-165698b05459.png)

At iteration 50 (output 10):
![image](https://user-images.githubusercontent.com/72091781/150187990-60ed10d9-e67b-4369-9437-2246bf9421b8.png)

As for performance, refining at frequency = 1 have the simulation plateau at about
`   Number of active cells:       5159
   Number of degrees of freedom: 16707
   Volume of triangulation:      140
   Number of VOF degrees of freedom: 5569
`

So it is notably quicker (on my laptop, with `mpirun -np 5`, each transient iteration takes about 5-10 sec). 

I could not launch your prm on my machine (10th gen i7, 16Go RAM), so I am concerned that users might find it impossible to launch this example as is. It will surely decrease a bit the agreement with the article, for this I think your prm is best as the mesh is really thin, but I think it would be better for documentation purposes to have an example launchable on a normal machine.

I will leave the final decision to the consensus!

Reviewer: The original prm file needs approximately 1 hour to finish on 8 cores. As this is not an application test, but an example, its goal is not to be fastly finished on laptops. We use examples to show how to set up simulations and show the accuracy of the solvers compared with experiments or other software results. I launched a simulation with the changes you made on the mesh and refinements. It's significantly faster, but still, with a fraction refinement = 0.9, the right edge of the interface is not covered with refined cells at several time-steps:
![image](https://user-images.githubusercontent.com/29518349/150191152-65ebde6f-0ae0-43ef-956e-1f56d60b0ac1.png)

I think this may cause problems when we use the post-processing code to obtain the maximum value of the interface since the post-processing code calculates the maximum value at `y` = 0. Hence, the blurry edge of the interface should lead to underestimation of the experimental values.

I agree to leave the final decision to the consensus.

Reviewer 3: It is not "fastly finished on laptops" with the changes I propose, it is still takes about 1 hour on my laptop with `mpirun -np 5`. I really agree that the agreement with the article will not be as good, obviously, but I would be concerned that the initial prm is not even launchable for a normal user (it was even stuck at initialization on my laptop... which is again pretty decent for computational work). 

Maybe the two could co-exist? Like, we provide a .prm that can be launched more easily by interested users, so that they can play around with the example, but we also give your prm and results after, to demonstrate the code abilities with a fine mesh and a good agreement with literature. This way we can also add a section to explain that for fast moving free surface, adaptation on the velocity catches well the interface at the front, but that for catching the interface globally it is faster to have adaptation on the phase. What do you think?

(sorry Abbas for the mess ^^" )

Reviewer: Very good idea.

Reviewer 3: @AbbasBalloutRe do not hesitate if you want me to help you rewriting the example!

Reviewer 3: This is the comparison with literature I get from the "light simulation" prm file :
![image](https://user-images.githubusercontent.com/72091781/150208148-7000dd28-5ba3-493b-8aa1-6910d49d2674.png)

It will be worth putting it also to explain the difference with the "accurate simulation" you put in place Shahab. This was already an impressive example, all of this together will make it super hot :)

Author: I just saw this one. Sorry

==========================
Dam break VOF
==========================

----------------------------------
Features
----------------------------------
- Solvers: ``gls_navier_stokes_2d`` 

> - Two phase flow handled by the Volume of fluids (VOF) approach with a phase fraction limiter and phase sharpening

Reviewer: ```suggestion
- Two phase flow handled by the Volume of Fluid (VOF) model with a phase fraction limiter and phase sharpening
```

> - Unsteady problem handled by an adaptive BDF1 time stepping scheme 

Reviewer: ```suggestion
- Unsteady problem handled by an adaptive BDF1 time-stepping scheme 
```

> - The use of a python script for post processing data

Reviewer: ```suggestion
- Data post-processing using a python (PyVista) script.
```

This example simulates the dam break experiments of `Martin and Moyce (1952)`_. 

.. _Martin and Moyce (1952): https://royalsocietypublishing.org/doi/abs/10.1098/rsta.1952.0006

A liquid is fixed behind a dam at the left most corner of
a rectangular domain as shown in the figure below.
At :math:`t = 0` s, the dam is removed, and 
the liquid is released into the total simulation domain. 
The corresponding parameter file is 
``gls_VOF_dam-break_Martin_and_Moyce.prm``.

The following schematic describes the geometry and dimensions of the simulation n the :math:`(x,y)` plane:

.. image:: images/VOF_dam_break_configuration.png
    :alt: Schematic
    :align: center
    :width: 400

.. note:: 
    All the four boundary conditions are ``slip``, and an external 
    gravity field of :math:`-1`` is applied in the y direction.

--------------
Parameter file
--------------

Time integration is handled by a 1st order backward differentiation scheme 

> `(bdf1)`, for a :math:`4.1` seconds simulation time with an initial 

Reviewer: ```suggestion
`(bdf1)`, for :math:`4.1` s simulation time with an initial 
```

time step of :math:`0.01` seconds.

.. note::   
    This example uses an adaptive time-stepping method in which the 
    time-steps are modified during the simulation to keep the maximum value of the CFL condition
    below a given threshold.

.. code-block:: text

    # --------------------------------------------------
    # Simulation Control
    #---------------------------------------------------
    subsection simulation control
        set method                         = bdf1
        set time end                       = 4.1
        set time step                      = 0.01
        set adapt                          = true
        set max cfl                        = 0.5
        set stop tolerance                 = 1e-5
        set adaptative time step scaling   = 1.1
        set number mesh adapt              = 0
        set output name                    = dam-break_VOF
        set output frequency               = 10
        set output path                    = ./Output/
        set subdivision                    = 1      
    end

.. warning::
    Make sure to create a directory named ``Output`` in the same directory 

>     you are calling the solver from.  Otherwise, the solver will be unable to generate the results files and will hang.

Reviewer: ```suggestion
    you are calling the solver from.  Otherwise, the solver will be unable to generate the results files and will break.
```

The ``multiphysics`` subsection enables to turn on `(true)` 
and off `(false)` the physics of interest. Here ``VOF`` and 
``interface sharpening`` are chosen.

.. note:: 
     If the interface sharpening is not selected, the interface 
     between phases will become blurry (due to diffusion). 
     The following section defines the interface sharpening 
     parameters.

.. code-block:: text

    #---------------------------------------------------
    # Multiphysics
    #---------------------------------------------------
    subsection multiphysics
        set VOF = true
        set interface sharpening = true
    end 

""""""""""""""""""""""""""""""""
Interface Sharpening Parameters
""""""""""""""""""""""""""""""""

The current ``interface sharpening`` method consists of two steps:

#. Phase fraction limiter   

    .. math:: 
        \phi := min \left( max \left(\phi^{old},0 \right),1 \right)
 

>     The phase fraction :math:`\phi` is a physical paramter that is bounded in the interval :math:`[0-1]`.

Reviewer 3: ```suggestion
The phase fraction :math:`\phi` is a physical paramter that is bounded in the interval :math:`[0-1]`.
```

>     The phase fraction limiter above will uopdate the phase fraction in case it fell outside of these bounds.

Reviewer 3: ```suggestion
The phase fraction limiter above will uopdate the phase fraction in case it fell outside of these bounds.
```

  

#. Interface sharpening 

    .. math::
        \phi :=
        \begin{cases}
        c^{1-\alpha} \phi^{\alpha} &  (0 \leq \phi < c  ) \\
        1-(c-1)^{1-\alpha}(1-\phi)^{\alpha} & (c \leq \phi \leq 1  ) 
        \end{cases}

``Sharpening frequency`` is an integer parameter that defines the 
frequency of the interface sharpening; ``sharpening threshold`` defines 
a phase fraction threshold for interface sharpening (generally :math:`0.5`);
and ``interface sharpness`` is a model parameter which is generally in
the range of :math:`(1-2]`.

>     where :math:`\phi`, :math:`c`, and :math:`\alpha` denote phase fraction, 

Reviewer 3: ```suggestion
where :math:`\phi`, :math:`c`, and :math:`\alpha` denote phase fraction, 
```

>     sharpening threshold, and interface sharpness respectively. 

Reviewer 3: ```suggestion
sharpening threshold, and interface sharpness respectively. 
```

>     This interface sharpening method was proposed by `Aliabadi and Tezduyar (2000)`_.  

Reviewer 3: ```suggestion
This interface sharpening method was proposed by `Aliabadi and Tezduyar (2000)`_.  
```

    .. _Aliabadi and Tezduyar (2000):  https://www.sciencedirect.com/science/article/pii/S0045782500002000

>     ``Sharpening frequency`` is an integer parameter that defines the 

Reviewer 3: ```suggestion
``Sharpening frequency`` is an integer parameter that defines the 
```

>     frequency of interface sharpening; sharpening threshold defines 

Reviewer 3: ```suggestion
frequency of interface sharpening; sharpening threshold defines 
```

>     a phase fraction threshold for interface sharpening (generally :math:`0.5`).

Reviewer 3: ```suggestion
a phase fraction threshold for interface sharpening (generally :math:`0.5`).
```

>     Interface sharpness is a model parameter which is generally in

Reviewer 3: ```suggestion
Interface sharpness is a model parameter which is generally in
```

>     the range of :math:`(1-2]`. 

Reviewer 3: ```suggestion
the range of :math:`(1-2]`. 
```

    end

We define two fluids here simply by setting the number of fluids to be :math:`2`.
In ``subsection fluid 0``, we set the density and the kinematic viscosity for the first phase. 
Similar procedure is done for the secondary phase in ``subsection fluid 1``. 

In the ``mesh adaptation subsection``, adaptive mesh refinement is 
defined for ``velocity``. ``min refinement level`` and ``max refinement 
level`` are 3 and 5, respectively.

.. code-block:: text

    #---------------------------------------------------
    # Mesh Adaptation
    #---------------------------------------------------
    subsection mesh adaptation
        set type                    = kelly
        set variable                = phase
        set fraction type           = fraction
        set max refinement level    = 5
        set min refinement level    = 3
        set frequency               = 1
        set fraction refinement     = 0.95
        set fraction coarsening     = 0.02
    end

Call the gls_navier_stokes_2d by invoking:  

``mpirun -np 2 gls_navier_stokes_2d gls_VOF_dam-break_Martin_and_Moyce.prm``

to run the simulation using two CPU cores. Feel free to use more.

.. warning:: 

>     The code will compute :math:`35,000+` dofs for :math:`600+` time 

Reviewer 3: change for the correct dof values (the prm has changed)

Author: Those are the correct ones

    iterations. Make sure to compile lethe in `Release` mode and 
    run in parallel using mpirun 

-------
Results
-------
The following image shows the screenshots 
of the simulation at :math:`0`, :math:`1.1`, :math:`3`, and :math:`4` seconds

.. image:: images/time-series.png
    :alt: time-shots
    :align: center

A python post-processing code `(Dambreak_2d_lethe.py)` 
is added to the example folder to post-process the results.

> Run `python3 ./Dambreak_2d_lethe.py ./Output` to execute this 

Reviewer 3: ```suggestion
Run ``python3 ./Dambreak_2d_lethe.py ./Output`` to execute this 
```

> post-processing code, where `./Output` is the directory that 

Reviewer 3: ```suggestion
post-processing code, where ``./Output`` is the directory that 
```

contains the simulation results. In post-processing, the maximum 
dimensionless lateral position of the liquid phase is tracked 
through time and compared with the experiments of Martin and Moyce
(1952). The following figure shows the result of

> the post-processing:

Reviewer 3: ```suggestion
the post-processing, with a very good agreement between the simulation and the experiment:
```

.. image:: images/xmax_t.png
    :alt: xmax_t
    :align: center

As mentioned previously, this simulation uses adaptive mesh
refinement. The following image shows the mesh and the position of

> the interface at :math:`4` seconds. The mesh refinement detects 

Reviewer: ```suggestion
the interface at :math:`4` s. The mesh refinement detects 
```

             set Function expression = if (x<3.5 & y<7 , 1, 0)
        end
    end

The ``source term`` subsection defines the gravitational acceleration:

.. code-block:: text
    
    #---------------------------------------------------
    # Source term
    #---------------------------------------------------
    subsection source term
        set enable = true
        subsection xyz
            set Function expression = 0;-1.0; 0
        end
    end

Two fluids are present in this simulation, hence in the ``physical 
properties`` subsection, their physical properties should be defined:

.. code-block:: text

    #---------------------------------------------------
    # Physical Properties
    #---------------------------------------------------
    subsection physical properties
        set number of fluids         = 2
        subsection fluid 0
            set density              = 0.02
            set kinematic viscosity  = 0.1
        end
        subsection fluid 1
            set density              = 0.9982
            set kinematic viscosity  = 0.01
        end
    end

We define two fluids here simply by setting the number of fluids to be :math:`2`.

> In ``subsection fluid 0``, we set the density and the kinematic viscosity for the first phase. 

Reviewer 2: ```suggestion
In ``subsection fluid 0``, we set the density and the kinematic viscosity for the first associated with a VOF indicator of 0.
```

> Similar procedure is done for the secondary phase in ``subsection fluid 1``. 

Reviewer 2: ```suggestion
Similar procedure is done for the phase associated with a VOF indicator of 1 in ``subsection fluid 1``. 
```

    end

We define two fluids here simply by setting the number of fluids to be :math:`2`.

> In ``subsection fluid 0``, we set the density and the kinematic viscosity for the first associated with a VOF indicator of 0. 

Reviewer 3: ```suggestion
In ``subsection fluid 0``, we set the density and the kinematic viscosity for the phase associated with a VOF indicator of 0. 
```

> 

Reviewer 3: Above, I would specify the color of the phases (sorry I did not say that earlier): (do not commit that as is, put this above)

```suggestion
The following image shows the screenshots of the simulation at :math:`0`, :math:`1.1`, :math:`3`, and :math:`4` seconds, with the red area and the blue area corresponding conversely to the water location and the air location.
```
